### DISCUSSION TITLE
[Feature]: Support for EXT-X-DATERANGE metadata

### DISCUSSION THREAD

**Discussion by pravind-hinge** [TheWidlarzGroup/react-native-video#4986] [Ideas]
### What problem are you trying to solve?

I'm trying to use an HLS playlist to stick 2 separate videos together like so:

```
#EXTM3U
#EXT-X-VERSION:6

#EXT-X-TARGETDURATION:25
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-PLAYLIST-TYPE:VOD
#EXT-X-INDEPENDENT-SEGMENTS

#EXT-X-DATERANGE:ID="intro",START-DATE="2023-01-01T00:00:00Z",DURATION=60.166666
#EXTINF:12.5,
/videos/project1/intro/1080p/video_000.ts
#EXTINF:8.333333,
/videos/project1/intro/1080p/video_001.ts
...

#EXT-X-DISCONTINUITY

#EXT-X-DATERANGE:ID="outro",START-DATE="2023-01-01T00:01:00.166666Z",DURATION=90.083334
#EXTINF:12.5125,
/videos/project1/outro/1080p/video_000.ts
...

#EXT-X-ENDLIST
```

I want to be able to get the exact time when the video switches from intro to outro. Using the `onProgress` event is not very consistent. So I was planning to use `EXT-X-DATERANGE`. However I'm not able to get the video player to trigger `onTimedMetadata`. As I'm using HLS format, embedding ID3 tags to `*.ts` is not working.

**Why it is needed?** To properly sync transitions UI with video progress dynamically and not rely on a video editor.

### Proposed solution

Support for `EXT-X-DATERANGE` metadata so that the player triggers `onTimedMetadata` (or an equivalent event) for HLS playlists using `EXT-X-DATERANGE` tags. An alternative approach with a priority event to sync UI should also be fine.

### Alternatives considered

- Using the `onProgress` event to detect the transition point, but this is not very consistent.
- Embedding ID3 tags in the `*.ts` segments to trigger `onTimedMetadata`, but this is not working with the HLS format.

[a comment by an automated account is not shown]

**Comment by TWG-AI** [comment]
Thanks for the idea — noted for the team to consider.

> I'm an AI assistant, so double-check any specifics I may have mentioned.
